### PR TITLE
[Android] Implement the feature about set accept languages.

### PR DESCRIPTION
**Pull request by wuhengzhi**
This patch is to implement the feature about set accept languages.
Preference registar was registered to "intl.accept_languages", if customized
accept languages was set to "intl.accept_languages", then invoke the sync
function to update accept languages.
setAcceptLanguages() was exposed to user to set the customized languages.

BUG=XWALK-3628, XWALK-3463

### CODE REVIEW

**Comment by wuhengzhi** [comment]
@lincsoon @axinging @halton Please help to review, thanks.

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContent.java [hunk 1]
```diff
@@ -182,6 +183,10 @@ protected void onReadyToRender() {
         // loaded by XMLHttpRequest.
         mSettings.setAllowFileAccessFromFileURLs(true);
 
+        String acceptLanguages = Locale.getDefault().toString();
```
Comment (halton) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkContent.java]: `acceptLanguages` -> `acceptLanguage` or `lang` 

`Locale.getDefault().toString()` -> `getDisplayLanguage()`

File: runtime/android/core_internal/src/org/xwalk/core/internal/XWalkSettings.java [hunk 2]
```diff
@@ -657,6 +658,33 @@ private void updateWebkitPreferencesOnUiThread() {
...
+                mAcceptLanguages = acceptLanguages;
+                mEventHandler.maybeRunOnUiThreadBlocking(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (mNativeXWalkSettings != 0) {
+                            nativeUpdateAcceptLanguages(mNativeXWalkSettings);
+                        }
+                    }
+                });
+            }
```
Comment (halton) [inline runtime/android/core_internal/src/org/xwalk/core/internal/XWalkSettings.java]: prefer
```
if (mAcceptLanguages == acceptLanguages) return;

mAcceptLanguages = acceptLanguages;
...
```

File: runtime/browser/android/xwalk_settings.cc [hunk 3]
```diff
@@ -232,6 +234,20 @@ void XWalkSettings::RenderViewCreated(
...
 }
 
+void XWalkSettings::UpdateAcceptLanguages(JNIEnv* env, jobject obj) {
+  ScopedJavaLocalRef<jstring> str =
+      Java_XWalkSettings_getAcceptLanguagesLocked(env, obj);
+
+  std::string accept_languages = base::android::ConvertJavaStringToUTF8(str);
+  PrefService* pref_service = GetPrefs();
+  if (!pref_service) return;
+  pref_service->SetString("intl.accept_languages", accept_languages);
```
Comment (halton) [inline runtime/browser/android/xwalk_settings.cc]: prefer

```
PrefService* pref_service = GetPrefs();
if (!pref_service) return;

pref_service->SetString(
    "intl.accept_languages",
    base::android::ConvertJavaStringToUTF8(
        Java_XWalkSettings_getAcceptLanguagesLocked(env, obj));
```

File: runtime/browser/xwalk_browser_context.cc [hunk 4]
```diff
@@ -67,12 +72,20 @@ class XWalkBrowserContext::RuntimeResourceContext :
   DISALLOW_COPY_AND_ASSIGN(RuntimeResourceContext);
 };
 
+XWalkBrowserContext* g_browser_context = nullptr;
+
+// Shows notifications which correspond to PersistentPrefStore's reading errors.
```
Comment (halton) [inline runtime/browser/xwalk_browser_context.cc]: I do not think this comment is valid, isn't it?

File: runtime/browser/xwalk_pref_store.cc [hunk 5]
```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2013 The Chromium Authors. All rights reserved.
```
Comment (halton) [inline runtime/browser/xwalk_pref_store.cc]: I think there is code change beside copy from chromium, if yes, please add Intel copyright

File: runtime/browser/xwalk_pref_store.cc [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+#include "base/memory/scoped_ptr.h"
+#include "base/values.h"
+
+XWalkPrefStore::XWalkPrefStore() {}
+
+XWalkPrefStore::~XWalkPrefStore() {}
+
+bool XWalkPrefStore::GetValue(const std::string& key,
+                           const base::Value** value) const {
```
Comment (halton) [inline runtime/browser/xwalk_pref_store.cc]: indent

File: runtime/browser/xwalk_pref_store.cc [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  return prefs_.GetValue(key, value);
+}
+
+bool XWalkPrefStore::GetMutableValue(const std::string& key,
+                                  base::Value** value) {
```
Comment (halton) [inline runtime/browser/xwalk_pref_store.cc]: indent

File: runtime/browser/xwalk_pref_store.h [hunk 8]
```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2013 The Chromium Authors. All rights reserved.
```
Comment (halton) [inline runtime/browser/xwalk_pref_store.h]: ditto

File: test/android/core/javatests/src/org/xwalk/core/xwview/test/SetAcceptLanuagesTest.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.xwalk.core.xwview.test;
+
+import android.test.suitebuilder.annotation.SmallTest;
+
+import org.chromium.base.test.util.Feature;
+
+
```
Comment (halton) [inline test/android/core/javatests/src/org/xwalk/core/xwview/test/SetAcceptLanuagesTest.java]: one empty line is enough

**Comment by wuhengzhi** [comment]
@halton Thanks a lot for your review. For ```Locale.getDefault().toString() -> getDisplayLanguage()```, we can not call ```getDisplayLanguage()``` since it's non-static function

**Comment by halton** [comment]
@wuhengzhi Unable to use `Locale.getDisplayLanguage()`? BTW, other part LGTM.

**Comment by wuhengzhi** [comment]
@halton Yes, it's non-static function, should be called by a instance of Locale.

**Comment by halton** [comment]
@wuhengzhi will merge.

**Pull request merged**